### DISCUSSION TITLE
Alternatives for yo generator

### DISCUSSION THREAD

**Discussion by albertoenriqueznews** [marianfoo/ui5-cc-excelUpload#623] [Q&A]
So I have a project that uses app directory with many UI apps inside them and Ive been looking on how to implement this library for 1 week and I wasnt able to find any example or any useful documentation yet.
The docs are very focused on using the yo generator, but thats not working in my workspace. 
What I did is looking for similar projects that use the latest version of this library on github but I barely found one.
What Im doing rn is: Instantiate the library in the manifest.json, npm install ui-cc-spreadsheetimporter on /app/myapp level, and then search on internet to see if I get the directory of the library "thirdparty/customcontrol/..." because yo generator is not working and I cant get the packages...
With this approach I could only find packages for 1.1.1 version max but I need for 1.2.0.
Is there any alternative of using the generator to get the packages?

**Comment by marianfoo** [comment]
> and then search on internet to see if I get the directory of the library "thirdparty/customcontrol/..." because yo generator is not working and I cant get the packages...

Why are you searching for this and why do you want to use the yo generator?  
And why are you focused on the package version from other apps in GitHub?

I don´t get what your actual problem is right now.

**Reply by albertoenriqueznews** [reply]
where do I search for the packages then? Maybe Im missing something from the docs, but I dont know how to get the packages. And by packages I mean the folder "thirdparty/..." that has everything that the library needs. I would appreciate if you let me know, thanks.

**Reply by marianfoo** [reply]
I still dont know what do you mean with "how to get packages". if you use it from npm you install it from npmjs 
https://www.npmjs.com/package/ui5-cc-spreadsheetimporter

**Reply by albertoenriqueznews** [reply]
Ill try to explain better. 
So I make an npm install on my app level inside /app directory. 
I copy these 2 lines to the manifest.json of that app: 
`"components": {
    "cc.spreadsheetimporter.v1_2_1": {}
  }
"componentUsages": {
    "spreadsheetImporter": {
        "name": "cc.spreadsheetimporter.v1_2_1"
    }
}`

Then, I go to my listReport controller and start using the sample functions that are in the docs. 
But nothing works:

`Uncaught (in promise) ModuleError: failed to load 'cc/spreadsheetimporter/v1_2_10/Component.js`
`failed to load JavaScript resource: cc/spreadsheetimporter/v1_2_10/Component-preload.js -  sap.ui.ModuleSystem`

What piece am I missing? Thanks

**Reply by marianfoo** [reply]
> v1_2_10

The version is `v1_2_1`

**Reply by marianfoo** [reply]
and you´re missing 

```
"resourceRoots": {
  "cc.spreadsheetimporter.v1_2_1": "./thirdparty/customcontrol/spreadsheetimporter/v1_2_1"
}
```

**Reply by marianfoo** [reply]
please read the docs carefully

**Reply by albertoenriqueznews** [reply]
Same issue.
Errors:
`ui5loader-dbg.js:1190 Uncaught (in promise) ModuleError: failed to load 'cc/spreadsheetimporter/v1_2_1/Component.js' from thirdparty/customcontrol/spreadsheetimporter/v1_2_1/Component.js`

`Uncaught (in promise) ModuleError: failed to load 'cc/spreadsheetimporter/v1_2_1/Component.js' from thirdparty/customcontrol/spreadsheetimporter/v1_2_1/Component.js`

app/myapp/package.json: 
`"dependencies": {
    "ui5-cc-spreadsheetimporter": "1.2.1"
  },`

app/myapp/webapp/manifest.json:
`"resourceRoots": {
      "cc.spreadsheetimporter.v1_2_1": "./thirdparty/customcontrol/spreadsheetimporter/v1_2_1"
    },
    "componentUsages": {
      "spreadsheetImporter": {
        "name": "cc.spreadsheetimporter.v1_2_1"
      }
    }`
